### DISCUSSION TITLE
Newbie developer wanting to contribute  first open source contribution

### DISCUSSION THREAD

**Discussion by ArpitSirohi** [jellyfin/jellyfin#9993] [General]
Hello ! I am a fresher .NET developer and have  never contributed to open source before . I looked at the goood first issues but they seem to be already taken. I wanted someone to recommend me a issue that would be appropriate for me and some guidance to me on how I can understand the jellyfin project . Thanks !!!

**Comment by joshuaboniface** [comment]
There aren't very many, but best bet would be to look for anything tagged "Help Wanted", or check over at the feature requests site (https://features.jellyfin.org) for something that seems interesting to you and work to implement it.

Or if anyone from @jellyfin/server has any suggestions please do!

**Reply by ArpitSirohi** [reply]
@joshuaboniface  Thanks for the advice . Help Wanted tags mostly seem to be taken but will try to take some other bug and try to resolve  if I can.

**Comment by mark-monteiro** [comment]
In case you missed it, there is also a full guide on contributing on the Jellyfin documentation site that I think would be useful to read through: https://jellyfin.org/docs/general/contributing/development. Hopefully that should answer any of your questions on _how_ to contribute.

If you're still wondering about _what_ to contribute, the issue lists or feature requests site that Joshua mentioned are a good place to start. Anecdotally, I started contributing by identifying something that was broken for me in the software that I wanted to fix and things sort of snowballed from there. Perhaps a good way to decide what to work on is to use the software a bunch and see if there are any areas that need fixing or improving for you personally.

I would also highly recommend joining the chat on Matrix (https://jellyfin.org/contact). There are lots of developers active there regularly and it can be helpful to bounce ideas around there. It can be especially helpful to check in with the chat before getting started on an issue, to make sure you're taking the right approach to the problem, nobody else is working on it, etc.

Hope that helps, good luck!

**Reply by ArpitSirohi** [reply]
Thanks for your advice Mark . Yes I did read the documentation it did give me more clarity .I will try to some issue that interests me and I believe can fix . I Have joined the Matrix chat also .
